src/pi_sonar.cpp

```diff
@@ -164,22 +183,44 @@ int main(int argc, char *argv[])
 
   ROS_INFO("Pi Sonar node ready");
 
+  diagnostic_updater::Updater updaters[5];
+  diagnostic_updater::HeaderlessTopicDiagnostic *pub_freq[5];
```

Comment:
Again move to sonar class, and use unique_ptr instead of raw ptr.

Reply:
Yes. I'll do that.